### PR TITLE
Dynamic reloading

### PR DESCRIPTION
**Pull request by karimnaaji**
Dynamically reload the scene when edits are made while authoring the map style.

### CODE REVIEW

**Comment by karimnaaji** [comment]
@matteblair I will leave this as is for a first review so we can discuss details of the PR for further future improvements.

File: Assets/Mapzen/Unity/Editor/MapStyleEditor.cs [hunk 1]
```diff
@@ -41,6 +44,16 @@ public override void OnInspectorGUI()
                 return;
             }
 
+            GUILayout.Label("Style hot reloading", labelBoldStyle);
```
Comment (matteblair) [inline Assets/Mapzen/Unity/Editor/MapStyleEditor.cs]: Maybe "Editing options" for this label?

File: Assets/Mapzen/Unity/Editor/MapStyleEditor.cs [hunk 2]
```diff
@@ -41,6 +44,16 @@ public override void OnInspectorGUI()
...
+
+            previewMode = GUILayout.Toggle(previewMode, "Preview mode");
```
Comment (matteblair) [inline Assets/Mapzen/Unity/Editor/MapStyleEditor.cs]: Let's just describe this option right on the button: "Update RegionMap while editing", then remove the space and label below this.

File: Assets/Mapzen/Unity/Editor/MapStyleEditor.cs [hunk 3]
```diff
@@ -17,6 +17,9 @@ public class MapStyleEditor : UnityEditor.Editor
         GUIStyle labelBoldStyle;
         GUIStyle labelItalicCenteredStyle;
 
+        [SerializeField]
+        private bool previewMode;
```
[the remark this answers is not in the document]
Comment (matteblair) [reply Assets/Mapzen/Unity/Editor/MapStyleEditor.cs]: Hmm actually the tree state seems to be reset when the inspector target changes too - I'll look into it.

[the remark this answers is not in the document]
Comment (matteblair) [reply Assets/Mapzen/Unity/Editor/MapStyleEditor.cs]: Ah I see, so an `EditorWindow` has some kind of serialized state, but an `Editor` only has the state of the target object. The examples for `TreeView` are all using `EditorWindow`s: https://docs.unity3d.com/Manual/TreeViewAPI.html

If we want to preserve these states we'd need them to be serialized in the target object (`MapStyle`) or make this into an `EditorWindow`.

**Pull request merged**